fix(java): Make source context work for obufscated filenames

romtsn opened the pull request:
Sometimes proguard/r8 can be used with the following option which will rename all of the filenames to `SourceFile`:
```
# If you keep the line number information, uncomment this to
# hide the original source file name.
-renamesourcefileattribute SourceFile
```

We were not considering that case, so we could end up with a wrong source filename, e.g. `io.sentry.samples.SourceFile` which doesn't exist in the linked source bundle. So this PR simply adds a check the filename should have a `.` in it, otherwise we consider it obfuscated and fall back to using the `module` field which will have the proper file path (by default JVM puts a source filename in this field, so it should always contain a file extension, see [the spec](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.7.10))

Closes https://github.com/getsentry/sentry-java/issues/3147

adinauer commented on tests/relay_integration/lang/java/test_plugin.py:
```diff
@@ -84,6 +84,12 @@ class AdditionalInnerClass {
     }
+
+    class OneMoreInnerClass {
+        fun whoops4() {
```
Not a big deal but this is never called anywhere in the code sample. Should we call it from `whoops3` above? Could be confusing for the next person coming along and trying to make sense of the example(s).

adinauer reviewed (approved):
Mostly LGTM, we could consider making the sample more realistic so it's not as confusing for people who try to make sense of it in the future.

romtsn commented:
Updated the tests, and also removed the if-else branch that is not gonna be hit, because we're checking for the dot presence earlier (also should make codecov happy): https://github.com/getsentry/sentry/pull/63588/files#diff-463f22f861cf56fa7b8dcf8326f098adf1f4a586ebbc741578801f7f840ff144L203-L206
